R/function_return_linter.R

```diff
@@ -28,6 +28,24 @@
 #'   linters = function_return_linter()
 #' )
 #'
+#' lint(
+#'   text = "foo <- function(x) return(x + 1); x <- x + 1",
```

Comment:
confused by this, maybe a typo? this is meant to be the "ok" equivalent of the <<- case right?

Reply:
Typo indeed. Fixed it.

Just wanted to show equivalent of achieving the same thing that doesn't produce a lint:

``` r
x <- 1
foo <- function(x) return(x <<- x + 1)
foo(x)
x
#> [1] 2

x <- 1
foo <- function(x) return(x + 1)
x <- foo(x)
x
#> [1] 2
```

<sup>Created on 2022-10-09 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>